### PR TITLE
feat: add overlay generation option for code transformations

### PR DESCRIPTION
**Pull request by diegosz**
Hi,

This change is only to have the option to apply transformations on a non empty folder.
It's just for using the generator as a library.

Many thanks for this awesome tool.
Cheers

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module github.com/rantav/go-archetype
+module github.com/diegosz/go-archetype
```
Comment (titusjaka) [inline go.mod:1]: Hi @diegosz,

Thank you for your contribution. I noticed that you renamed the module to point to your repository. Could you please revert this change?

Comment (diegosz) [reply go.mod:1]: oops, sorry about that

Comment (diegosz) [reply go.mod:1]: Please close this PR, already made this one: https://github.com/rantav/go-archetype/pull/29

**Comment by diegosz** [comment]
Change to this PR: https://github.com/rantav/go-archetype/pull/29
